### PR TITLE
Fix .NET 5 FileStream compat detection in the test utility project

### PR DESCRIPTION
**Pull request by adamsitnik**
`FileStreamHelpers` was moved from `System.IO` to `System.IO.Strategies` namespace

https://github.com/dotnet/runtime/blob/a4894885a534c3c9091fd7dedae18d2686c6b844/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.cs#L7-L9

### CODE REVIEW

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 1]
```diff
@@ -286,7 +286,7 @@ private static Version GetICUVersion()
                               version & 0xFF);
         }
 
-        private static readonly Lazy<bool> _net5CompatFileStream = new Lazy<bool>(() => GetStaticNonPublicBooleanPropertyValue("System.IO.FileStreamHelpers", "UseNet5CompatStrategy"));
+        private static readonly Lazy<bool> _net5CompatFileStream = new Lazy<bool>(() => GetStaticNonPublicBooleanPropertyValue("System.IO.Strategies.FileStreamHelpers", "UseNet5CompatStrategy"));
```
Comment (stephentoub) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:289]: Should we make this throw if the type can't be found?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs (hunk 1 above)
Comment (adamsitnik) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:289]: I was thinking about it, but that would require special handling for when the `PlatformDetection` type is used to run Full .NET Framework tests. I decided to just add a test instead. Is that OK?

Comment (carlossanlop) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:289]: I think Adam meant (please correct me if I'm wrong) that the `PlatformDectection` class, which contains the `_net5CompatFileStream` field, is also consumed by some .NET Framework tests. I can see the file can be built for both net60 and net461:
image

What I do not know is what special handling would be required to make the exception work.

**Review by carlossanlop — APPROVED** [review]
LGTM. Pending answering the question posted by Stephen.

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs (hunk 1 above)
Comment (adamsitnik) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:289]: I've forgotten that it's lazy and only compiled for .NET 4.6.1 (not executed).

**Pull request merged**